src/core/keybindings.c

```diff
@@ -1321,17 +1317,15 @@ meta_display_process_key_event (MetaDisplay *display,
 
 #ifdef HAVE_XKB
   keysym = XkbKeycodeToKeysym (display->xdisplay, event->xkey.keycode, 0, 0);
-  str = XKeysymToString (keysym);
 #else
   keysym = 0;
-  str = NULL;
 #endif
 
   /* was topic */
   meta_topic (META_DEBUG_KEYBINDINGS,
               "Processing key %s event, keysym: %s state: 0x%x window: %s\n",
               event->type == KeyPress ? "press" : "release",
-              str ? str : "none", event->xkey.state,
+              XKeysymToString (keysym) ? XKeysymToString (keysym) : "none", event->xkey.state,
```

Comment:
Same question, but it's worse here: `keysym` has a fake value of `0` in the `! HAVE_KXB` code path, what would be the return value of `XKeysymToString(0)`?  wouldn't it be a tad confusing?